### PR TITLE
docs: Type-Level Programming examples (related to #6062)

### PR DESCRIPTION
**Pull request by xdesou01**
WIP

### CODE REVIEW

File: main/src/library/TLP.flix [hunk 1]
```diff
@@ -0,0 +1,194 @@
+//Example 1:
```
Comment (magnus-madsen) [inline main/src/library/TLP.flix:1]: Lets split this into three files and place them into `examples/`

File: examples/type-level-porgramming-victuals.flix [hunk 2]
```diff
@@ -0,0 +1,101 @@
+//An interface that involves possibly infected foodstuffs.
```
Comment (magnus-madsen) [inline examples/type-level-porgramming-victuals.flix:1]: A few comments about code style

File: examples/type-level-porgramming-victuals.flix [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+//Note: Please follow conventionally accepted culinary practices IRL
```
Comment (magnus-madsen) [inline examples/type-level-porgramming-victuals.flix:2]: Comments should use triple slashes `/// foo` and with a space.

File: examples/type-level-porgramming-victuals.flix [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+enum Victual[_infection: Bool] {
+    case F({victual_name = String, weight = Int32, disease = Option[String]})
```
Comment (magnus-madsen) [inline examples/type-level-porgramming-victuals.flix:4]: fields are camelCase not snake_case.

File: examples/type-level-porgramming-victuals.flix [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+def poison(v: Victual[Safe], d: String): Victual[Infected] = match v {
+    case Victual.F(r) =>
+      Victual.F({victual_name = r.victual_name, weight = r.weight, disease = Some(d)})
+}
+
+//You may wish to make a meal by mixing two existing victuals.
+
+
+//helper function for nomenclature modularity
+def new_name_and_weight(n1: String, n2: String, w1: Int32, w2: Int32): (String, Int32) = {
```
Comment (magnus-madsen) [inline examples/type-level-porgramming-victuals.flix:36]: function names are also camelCase, e.g. newNameAndWeight

**Review by magnus-madsen — COMMENTED** [review]
I added some comments on code style you can fix in all three examples.

**Comment by magnus-madsen** [comment]
Do we have more good examples or are we done? I guess we never tried any board games.

**Comment by xdesou01** [comment]
> Do we have more good examples or are we done? I guess we never tried any board games.

That is all for now. I can begin working on some more if you would like though

**Comment by magnus-madsen** [comment]
I think we merged in some reasonable examples earlier in different PRs.

**Pull request closed without merging**